src/Surfnet/StepupGateway/SecondFactorOnlyBundle/Service/Gateway/SecondFactorLoginService.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/**
+ * Copyright 2018 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupGateway\SecondFactorOnlyBundle\Service\Gateway;
+
+use Psr\Log\LoggerInterface;
+use Surfnet\SamlBundle\Http\HttpBindingFactory;
+use Surfnet\SamlBundle\Monolog\SamlAuthenticationLogger;
+use Surfnet\SamlBundle\SAML2\AuthnRequest;
+use Surfnet\SamlBundle\SAML2\ReceivedAuthnRequest;
+use Surfnet\StepupGateway\SecondFactorOnlyBundle\Service\LoaResolutionService;
+use Surfnet\StepupGateway\GatewayBundle\Exception\RequesterFailureException;
+use Surfnet\StepupGateway\GatewayBundle\Saml\Proxy\ProxyStateHandler;
+use Surfnet\StepupGateway\SecondFactorOnlyBundle\Service\SecondFactorOnlyNameIdValidationService;
+use Symfony\Component\HttpFoundation\Request;
+
+class SecondFactorLoginService
```

Comment:
'Only' is missing from the service name right? Or did you name this class intentionally?

Reply:
Changed